Data/HashMap/Strict.hs

```diff
@@ -506,6 +618,6 @@ updateOrSnocWithKey f k0 v0 ary0 = go k0 v0 ary0 0 (A.length ary0)
 -- These constructors make sure the value is in WHNF before it's
 -- inserted into the constructor.
 
-leaf :: Hash -> k -> v -> HashMap k v
+leaf :: Hash -> k -> v -> HashMapInner k v
```

Comment:
At risk of being nit-picky, I'd pick a simpler name like `Tree` over `HashMapInner` for this internal data structure, given how often it's used internally.

Reply:
Not nitpicky at all, that's a good point.